neo/SmartContract/InteropService.cs

```diff
@@ -145,53 +145,31 @@ internal static bool CheckWitness(ApplicationEngine engine, UInt160 hash)
         {
             if (engine.ScriptContainer is Transaction tx)
             {
-                // will return false, unless some scope matches witness requirement
-                var ret = false;
-
-                foreach (var attribute in tx.Attributes)
+                CosignerUsage usage = tx.Attributes
+                    .Where(p => p.Usage == TransactionAttributeUsage.Cosigner)
+                    .Select(p => p.Data.AsSerializable<CosignerUsage>())
+                    .FirstOrDefault(p => p.Account.Equals(hash));
+                if (usage is null) return false;
```

Comment:
If there are more than one? is not better to check all of them?

Reply:
https://github.com/neo-project/neo/blob/53f7d022afba7b1dfce5a437f8bc11557423a7fb/neo/Network/P2P/Payloads/Transaction.cs#L95-L96

In fact it is impossible.

Reply:
But here you filter by account, and there is filtered by the whole data